### PR TITLE
Added code for integration of events

### PR DESCRIPTION
**Pull request by cyble-dev**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by thefrieddan1** [comment]
Hi @cyble-dev, 
Thank you for your contribution.
In order for me to start reviewing, 
please be sure to fill the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA)

Thanks.

**Review by RotemAmit — CHANGES_REQUESTED** [review]
Nice work!
* Please run demisto-sdk pre-commit and fix the validate errors, if there are any.
* Add a readme to the playbook.
* Since you are creating a v2 to CybleEvents integration, please deprecate the old version. here is a link to an article on how to do it https://xsoar.pan.dev/docs/reference/articles/deprecation-process-and-hidden-packs#how-to-deprecate-an-integration.
* Since the old version of the integration will be deprecated, please add the command 'cyble-vision-fetch-event-detail' that is on v1 but not in v2 in order not to break backword compatibility.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/README.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- | --- |
+    | URL | Server URL \(e.g. https://example.net\) | True |
+    | Access Token | Access Token | True |
+    | Trust any certificate (not secure) |  | False |
+    | Use system proxy settings |  | False |
+    | Incidents Fetch Interval |  | False |
+    | Incident Fetch Limit | Maximum incidents to be fetched every time. Upper limit is 50 incidents. | False |
+
```
[the remark this answers is not in the document]
Comment (cyble-dev) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/README.md:18]: @RotemAmit , Is First fetch time is mandatory for Pack (Integration). if its mandatory, Kindly let me know, how to add them in the implementation

[the remark this answers is not in the document]
Comment (RotemAmit) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/README.md:18]: The parameter first fetch time, isn't mandatory but it should has a default value that you should use in the first run of the fetch, When demisto.getLastRun() = None

[the remark this answers is not in the document]
Comment (cyble-dev) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/README.md:18]: If data is none, I am assigning the current date in the last run, below code is added in line 358, is this fine

 if 'event_pull_start_date' not in last_run.keys():
            last_run['event_pull_start_date'] = datetime.now().astimezone().replace(microsecond=0).isoformat()

[the remark this answers is not in the document]
Comment (RotemAmit) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/README.md:18]: Please change this assignment and use there the first fetch time.

**Comment by efelmandar** [comment]
Hi @cyble-dev  first of all thank you for your contribution. I reviewed the content in your PR and found a couple of points I would like to bring to your attention.

1. In the `Incident Type`, the layout field should be the layout name, not an ID.
image

2. In the playbook, I see that it only contains a single task of fetch alert, isn't it the same as `fetch incidents` in the integration? If it's not the same, I would consider adding other things to this playbook to bring the analyst more data such as enrichment maybe.

3. Also in the playbook if you decide to keep it, you pass the start, end date and limit as static values, I would recommend changing the task to use playbook inputs instead and setting the static dates and limit as default values for these inputs.
image

Please feel free to reach out if you have any questions or need any additional help, here or on DFIR channel on Slack.

**Comment by efelmandar** [comment]
@cyble-dev Thank you for making the suggested changes and I approve the PR from my side.

Thank you again for your hard work and contribution to our platform.

**Review by RotemAmit — COMMENTED** [review]
Good work!
Here are some notes:
* Update the README files with the new changes in the YML and add the sections Command Example and Context Example, as in the example and explanations in this [link](https://xsoar.pan.dev/docs/documentation/readme_file#documentation-examples).
* Please bump the version of the pack CybleThreatIntel in the pack_metadata.json file and add notes describing your changes to a new Release Notes file. 
To do that, please run the `demisto-sdk update-release-notes` command - for more information please see: https://xsoar.pan.dev/docs/concepts/demisto-sdk#update-release-notes.

**Review by RotemAmit — COMMENTED** [review]
Please resolve all the conflicts in CybleThrearIntel pack and in the pack metadata file of CybleEvents pack.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    - default: false
+      description: Timeline end date in the format "%Y-%m-%dT%H:%M:%S%z" (iso-8601)
+      isArray: false
+      name: endDate
+      required: true
+      secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: asc
+      description: Sorting order for alert fetch either Ascending or Descending
```
[the remark this answers is not in the document]
Comment (RotemAmit) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml:171]: Please do this change.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+      secret: false
+    - default: false
+      description: Timeline end date in the format "%Y-%m-%dT%H:%M:%S%z"  (iso-8601)
+      isArray: false
+      name: endDate
+      required: true
+      secret: false
+    - default: false
+      defaultValue: '0'
+      description: Returns records for the timeline starting from given indice
```
[the remark this answers is not in the document]
Comment (RotemAmit) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml:236]: Please do this change.

**Review by RotemAmit — COMMENTED** [review]
Nice work!
Please make sure to resolve the remaining comments from the previous reviews as well.

File: Packs/CybleEventsV2/README.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
+# Cyble Events
```
Comment (RotemAmit) [inline Packs/CybleEventsV2/README.md:1]: Please update this file in order for it to match the structure in this [link](https://xsoar.pan.dev/docs/documentation/pack-docs#pack-readme) to xsoar pan dev website.

**Review by RotemAmit — COMMENTED** [review]
Nice work!
Please update the docker image tag of the integration CybleThreatIntel. You can check for the most updated version of demisto/taxii-server here: https://hub.docker.com/r/demisto/taxii-server/tags. After you commit this change please run demisto-sdk update-release-notes, so this change will be added to the release notes file.

**Review by RotemAmit — COMMENTED** [review]
Nice work!
Please update the docker image tag of the integration CybleThreatIntel. You can check for the most updated version of demisto/taxii-server here: https://hub.docker.com/r/demisto/taxii-server/tags. After you commit this change please run demisto-sdk update-release-notes, so this change will be added to the release notes file.

**Comment by efelmandar** [comment]
Hi @cyble-dev Thank you for all the hard work, I still see some validation errors which need to be fixed in order for the PR to pass, 

1. In the Incident Type, the layout field should have the layout name, not an ID. In your case, for example, it should be
`"layout": "Cyble Vision Alert V2"` instead of `"layout": "91b2b1be-7c2e-4ec5-8be8-3407aba92b65"`

2. In the Incident Type, I also see some mixup where the filename is `Cyble Inter Alert V2` but the name and ID fields in the JSON are `Cyble Vision Alert V2`

3. The playbook validation is failing because in the playbook you reference the integration ID (`cybleeventsv2`) but in the integration YML you've added a space in the ID and name `cybleevents V2` so to fix it please remove the space from the ID and Name fields in the integration YML

File: Packs/CybleThreatIntel/Integrations/CybleThreatIntel/CybleThreatIntel.yml [hunk 5]
```diff
@@ -136,7 +135,7 @@ script:
     - contextPath: CybleIntel.collection.names
       description: Available collection names for the feed service
     description: Get the data feed collection names for the taxii feed
-  dockerimage: demisto/taxii-server:1.0.0.61738
+  dockerimage: demisto/taxii-server:1.0.0.56695
```
Comment (RotemAmit) [inline Packs/CybleThreatIntel/Integrations/CybleThreatIntel/CybleThreatIntel.yml:138]:
```suggestion
  dockerimage: demisto/taxii-server:1.0.0.62984
```

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+      secret: false
+    deprecated: false
+    description: Fetch incident event group
+    execution: false
+    name: cyble-vision-fetch-alert-groups
+    outputs:
+    - contextPath: CybleEvents.AlertGroup
+      description: Fetch all the alert groups
+      type: String
+  dockerimage: demisto/python3:3.10.11.61265
```
Comment (RotemAmit) [inline Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml:249]:
```suggestion
  dockerimage: demisto/python3:3.10.12.62631
```

**Review by RotemAmit — COMMENTED** [review]
Nice work!
* Please in the release notes of CybleThreatIntel use the same amount of # as I asked.
* Please run: demisto-sdk format -i Packs/CybleThreatIntel/Integrations/CybleThreatIntel/CybleThreatIntel.yml

**Review by RotemAmit — COMMENTED** [review]
Nice work! Thank you very much for your contribution!

File: Packs/CybleThreatIntel/Integrations/CybleThreatIntel/CybleThreatIntel.yml [hunk 7]
```diff
@@ -136,7 +136,7 @@ script:
     - contextPath: CybleIntel.collection.names
       description: Available collection names for the feed service
     description: Get the data feed collection names for the taxii feed
-  dockerimage: demisto/taxii-server:1.0.0.61738
+  dockerimage: demisto/taxii-server:1.0.0.62984
```
Comment (RotemAmit) [inline Packs/CybleThreatIntel/Integrations/CybleThreatIntel/CybleThreatIntel.yml:139]:
```suggestion
  dockerimage: demisto/taxii-server:1.0.0.63049
```

File: Packs/CybleThreatIntel/ReleaseNotes/2_0_11.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+#### Integrations
+##### Cyble Threat Intel
+- Updated the Docker image to: *demisto/taxii-server:1.0.0.62984*.
```
Comment (RotemAmit) [inline Packs/CybleThreatIntel/ReleaseNotes/2_0_11.md:3]:
```suggestion
- Updated the Docker image to: *demisto/taxii-server:1.0.0.63049*.
```

**Review by RotemAmit — COMMENTED** [review]
Please Update the docker image as in the comments

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+      secret: false
+    deprecated: false
+    description: Fetch incident event group
+    execution: false
+    name: cyble-vision-fetch-alert-groups
+    outputs:
+    - contextPath: CybleEvents.AlertGroup
+      description: Fetch all the alert groups
+      type: String
+  dockerimage: demisto/python3:3.10.12.63049
```
Comment (RotemAmit) [inline Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml:248]:
```suggestion
  dockerimage: demisto/python3:3.10.12.62631
```

**Review by RotemAmit — APPROVED** [review]

**Pull request merged**